src/pages/write-smart-contracts/billboard-tutorial.md

```diff
@@ -0,0 +1,281 @@
+the `cur-price`, which represents the current price of updating the billboard, and the `new-price`, which represents the
+incremented price for updating the billboard.
+
+The function then calls the [`stx-transfer?`][] function to transfer the current price of the contract in STX from the
+transaction sender to the contract wallet. This syntax can be confusing: the function call uses the `tx-sender`
+variable, which is the principal address of the caller of the function. The second argument to [`stx-transfer?`][] uses
+the [`as-contract`][] function to change the context's `tx-sender` value to the principal address that deployed the
+contract.
+
+The entire [`stx-transfer?`][] function call is wrapped in the [`unwrap!`][] function, to provide protection from
+the transfer failing. The [`unwrap!`][] function executes the first argument, in this case the [`stx-transfer?`][]
+function. If the execution returns `(ok ...)`, the [`unwrap!`][] function returns the inner value of the `ok`, otherwise
+the function returns the second argument and exits the current control-flow, in this case the `ERR_STX_TRANSFER` error
+code.
+
+If the token transfer is successful, the function sets the new `billboard-message` and updates the `price` variable to
+`new-price`. Finally, the function returns `(ok new-price)`. It's generally a good practice to have public functions
+return `ok` when successfully executed.
+
+```clarity
+(define-public (set-message (message (string-utf8 500)))
+    (let ((cur-price (var-get price))
+          (new-price (+ cur-price u10)))
+
+        ;; pay the contract
+        (unwrap! (stx-transfer? cur-price tx-sender (as-contract tx-sender)) (err ERR_STX_TRANSFER))
+
+        ;; update the billboard's message
+        (var-set billboard-message message)
+
+        ;; update the price
+        (var-set price new-price)
+
+        ;; return the updated price
+        (ok new-price)
+    )
+)
+```
+
+At this point, the final contract should look like this:
+
+```clarity
+;; error consts
+(define-constant ERR_STX_TRANSFER u0)
+
+;; data maps/vars
+(define-data-var billboard-message (string-utf8 500) u"Hello World!")
+(define-data-var price uint u100)
+
+;; public functions
+(define-read-only (get-price)
+    (var-get price)
+)
+
+(define-read-only (get-message)
+    (var-get billboard-message)
+)
+
+(define-public (set-message (message (string-utf8 500)))
+    (let ((cur-price (var-get price))
+          (new-price (+ cur-price u10)))
+
+        ;; pay the contract
+        (unwrap! (stx-transfer? cur-price tx-sender (as-contract tx-sender)) (err ERR_STX_TRANSFER))
+
+        ;; update the billboard's message
+        (var-set billboard-message message)
+
+        ;; update the price
+        (var-set price new-price)
+
+        ;; return the updated price
+        (ok new-price)
+    )
+)
+```
+
+## Step 5: write a contract test
```

Comment:
Maybe it would be helpful to run `clarity check` here to ensure devs got the contract pasted correctly before moving to the tests?